### ISSUE TITLE
OSError raised on many concurrent reads

### ISSUE THREAD

**Issue by d-chambers** [niosh-mining/obsplus#69]
* obsplus version:
0.0.1
* Python version:
3.7
* Operating System:
Ubuntu 16.04

### Description

When many processes are accessing the same `WaveBank` occasionally a traceback ending in the following is issued:

```
OSError: HDF5 error back trace

  File "H5F.c", line 511, in H5Fopen
    unable to open file
  File "H5Fint.c", line 1519, in H5F_open
    unable to lock the file
  File "H5FD.c", line 1650, in H5FD_lock
    driver lock request failed
  File "H5FDsec2.c", line 941, in H5FD_sec2_lock
    unable to lock file, errno = 11, error message = 'Resource temporarily unavailable'

End of HDF5 error back trace
```
This is originating from a call to `pd.read_hdf(self.index_path, self._meta_node)`. I suspect this may be happening because another process has the HDF5 file open in write mode (in order to update the index) at the same time this call is made.

**Comment by d-chambers** [comment]
This seems related: https://github.com/pandas-dev/pandas/issues/4409

**Comment by d-chambers** [comment]
closed by #70
